lib/Runtime/Types/RecyclableObject.h

```diff
@@ -292,7 +292,7 @@ namespace Js {
         virtual PropertyId GetPropertyId(BigPropertyIndex index) { return Constants::NoProperty; }
         virtual PropertyIndex GetPropertyIndex(PropertyId propertyId) { return Constants::NoSlot; }
         virtual int GetPropertyCount() { return 0; }
-        virtual PropertyQueryFlags HasPropertyQuery(PropertyId propertyId);
+        virtual PropertyQueryFlags HasPropertyQuery(PropertyId propertyId, _Inout_opt_ PropertyValueInfo* info = nullptr);
```

Comment:
> = nullptr [](start = 110, length = 10)

Is the default really needed? It means an omission to pass info through (as in many places you had to do) cannot be caught by the compiler. If there are only few calls when default is used, it could be safer to updated them to pass nullptr explicitly.

Reply:
I tried removing the default yesterday to see how many places were using the default, and I don't remember the number but it was large enough that I didn't feel very motivated to change it. I believe that failures to pass info through are generally not correctness problems, but might prevent something from being cached when it could be. I don't feel very strongly either way on this, so I can remove the default if you prefer.

Reply:
I don't feel strongly about it either, just wondered if it would be worth the trouble.

---
In reply to: [175184369](https://github.com/Microsoft/ChakraCore/pull/4834#discussion_r175184369) [](ancestors = 175184369)